src/character.cpp

```diff
@@ -2550,6 +2550,45 @@ std::string Character::extended_description() const
     return replace_colors( ss.str() );
 }
 
+//@todo the interaction type should either be an enum or jsonized
+int Character::get_mutation_social_mod( const std::string &type ) const
+{
+    int mod = 0;
+    for( const mutation_branch *mut : cached_mutations ) {
+        int mut_mod = 0;
+        for( const mut_social_mod &soc_mod : mut->social_mods ) {
+            // If player has any blocker, bail out
+            if( std::any_of( soc_mod.blocker_mutations.begin(), soc_mod.blocker_mutations.end(),
+                [this]( const trait_id &blocker ) {
+                    return has_trait( blocker );
+                } ) ) {
+                continue;
+            }
+
+            // Player must have all needed traits
+            if( !std::all_of( soc_mod.required_mutations.begin(), soc_mod.required_mutations.end(),
+                [this]( const trait_id &need ) {
+                    return has_trait( need );
+                } ) ) {
+                continue;
+            }
+
+            if ( type == "lie" ) {
+                mut_mod += soc_mod.lie;
+            } else if ( type == "persuade" ) {
```

Comment:
Using `else if` means that we can't get modifiers for several types, e.g. `lie` AND `persuade`, or I'm getting it wrong?

Also, astyle.

Reply:
I can't think of any circumstances when you'd need to more than one.  I could change it to just return a struct of all the values at once, though.

Come to think of it, that would get rid of the ugly string types as well